File: app/src/main/java/com/saint/struct/activity/BaseActivity.java
package com.saint.struct.activity;

import android.os.Bundle;

import com.saint.struct.tool.StatusBarUtils;

import androidx.annotation.Nullable;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;

public class BaseActivity extends AppCompatActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        hideActionBar();

        StatusBarUtils.setDeepStatusBar(true, this);
    }

    public void hideActionBar(){
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.hide();
        }
    }
}

File: app/src/main/java/com/saint/struct/activity/BaseActivity.kt
package com.saint.struct.activity

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.saint.struct.tool.StatusBarUtils

open class BaseActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        hideActionBar()
        StatusBarUtils.setDeepStatusBar(true, this)
    }

    fun hideActionBar() {
        val actionBar = supportActionBar
        actionBar?.hide()
    }
}